build/docker/intel-deviceplugin-operator.Dockerfile

```diff
@@ -19,6 +19,7 @@
 # This is used on release branches before tagging a stable version.
 # The main branch defaults to using the latest Golang base image.
 ARG GOLANG_BASE=golang:1.17-bullseye
+ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro
```

Comment:
Just like we have with `GOLANG_BASE`, we want `FINAL_BASE` documented. How about:

```suggestion
# FINAL_BASE can be used to configure the base image  of the final image.
#
# This is used in two ways:
# 1) make <image-name> BUILDER=<docker|buildah>
# 2) docker build ... -f <image-name>.Dockerfile
#
# The project default is 1) which sets FINAL_BASE=gcr.io/distroless/static
# (see build-image.sh). 
ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro
```

I wonder would it be possible to toggle the default and override the value in that Redhat build service?

Reply:
The documentation for FINAL_BASE looks good. Will add it. However, I am afraid we cannot keep the default base and configure the RedHat build service. There is no option to pass any extra arguments to the build service.

Reply:
@mythi It doesn't look good to me to change our defaults like that to be honest.

Reply:
what changed your mind? we discussed this earlier and it was OK.

Reply:
Understanding the fact that default build would not use our default base image changed my perception of this.

Reply:
@bart0sh yeah I agree on that part. let's go with this now. I'll contribute the tests to make sure we always build/push distroless based images.

@chaitanya1731 one additional review comment was requested offline: Let's also add in the comments when and why the default gets used: "2) and the default FINAL_BASE is primarily used to build Redhat Certified Openshift Operator images that must be UBI based. The build tool does not allow additional image build parameters."

Reply:
@chaitanya1731 can you explain how the RedHat build service is going to build our images? Is it possible to use Makefile targets or it just finds Dockerfiles and runs ```docker build``` for them directly?

Reply:
https://redhat-connect.gitbook.io/partner-guide-for-red-hat-openshift-and-container/certify-your-application/creating-a-container-application-project

Reply:
@mythi Thanks. I don't see any information about Dockerfile usage there for some reason.

Reply:
Two clicks ahead talks about the Dockerfiles: https://redhat-connect.gitbook.io/partner-guide-for-red-hat-openshift-and-container/certify-your-application/image-upload

Reply:
@chaitanya1731 thank you for the info. And how we're going to configure those builds? Are we going to point them out to our source tree somehow or just copy&paste our Dockerfiles somewhere?

Reply:
@bart0sh We mention git source url, branch name, context directory and the path to the dockerfile. 
Please see the image attached.

![image](https://user-images.githubusercontent.com/14907392/155370648-47b1a9b0-bd2a-4027-aafa-d1eace884577.png)

Reply:
@chaitanya1731 thank you for the explanations. Now I start to understand how it works.
@mythi would it make sense to set FINAL_BASE=<our default image> in all Dockerfiles and generate a separate set of Dockerfiles (with some suffix or something) for this purpose on every merge by setting FINAL_BASE to registry.access.redhat.com/ubi8-micro?